actions/locations.js

```diff
@@ -175,6 +176,8 @@ export function pushLocalLocations() {
       dispatch(wrapper(saved, location));
     }));
 
+    setTimeout(() => dispatch(setUploadingStatus(false)), 2000);
```

Comment:
Why do we need this? Aren't failed promises enough?

Reply:
The UploadingStatus is used to tell when we are uploading locations as a whole not for individual locations. I wanted this so that I can update the UI accordingly when we are uploading data and then when the data is finished uploading. So a failed promise would just tell me that one of the locations failed but I need to wait till they are all done failed or otherwise.

Reply:
@smaclell Maybe I am missing something but I'm not sure I understand how adding this requires us to catch the failures individually.  Since we are awaiting the Promise.all won't all of the locations be completed regardless of whether some of them failed or not?

All the added line cares about is if they are done, not if they succeeded or not.

Reply:
My thought is the difference between setTimeout vs waiting for the promises to finish. The artificial timeout doesn't really make sense to me. I don't know if promise.all waits for everything if one fails

Reply:
Ahh yeah, I should have explained the setTimeout. The setTimeout is only there because if you only upload like a couple location they complete really quickly and the UI doesn't update quick enough. This just ensures the UI will show the uploading state on the screen for at least two seconds.

Reply:
oh!!! mmm ... that is weird. Is there a pure component where maybe there shouldn't be?

Reply:
Haha, yup that was it, good catch!

Reply:
I don't think it has to do with that. It's more just the fact that If you have say only one location to upload the UI on the new screen won't even update before the location finishes being pushed.

I got rid of the PureComponent and it didn't help.

Reply:
Still need to chat about it ...